cudamapper/src/overlapper.cpp

```diff
@@ -98,6 +98,51 @@ bool overlaps_mergable(const claraparabricks::genomeworks::cudamapper::Overlap o
     return short_gap_relative_to_length;
 }
 
+inline bool overlaps_identical(const claraparabricks::genomeworks::cudamapper::Overlap& a, const claraparabricks::genomeworks::cudamapper::Overlap& b)
+{
+    return a.query_read_id_ == b.query_read_id_ &&
+           a.target_read_id_ == b.target_read_id_ &&
+           a.query_start_position_in_read_ == b.query_start_position_in_read_ &&
+           a.query_end_position_in_read_ == b.query_end_position_in_read_ &&
+           a.target_start_position_in_read_ == b.target_start_position_in_read_ &&
+           a.target_end_position_in_read_ == b.target_end_position_in_read_;
+}
+
+inline double percent_reciprocal_overlap(const claraparabricks::genomeworks::cudamapper::Overlap& a, const claraparabricks::genomeworks::cudamapper::Overlap& b)
+{
+    if (a.query_read_id_ != b.query_read_id_ || a.target_read_id_ != b.target_read_id_ || a.relative_strand != b.relative_strand)
+    {
+        return 0.0;
+    }
+    int32_t query_overlap = std::min(a.query_end_position_in_read_, b.query_end_position_in_read_) - std::max(a.query_start_position_in_read_, b.query_start_position_in_read_);
```

Comment:
This and similar should be `position_in_read_t` or even better `number_of_basepairs_t`